app/models/chargeback_rate_detail.rb

```diff
@@ -92,4 +92,22 @@ def rate_type
     # Return parent's rate type
     chargeback_rate.rate_type unless chargeback_rate.nil?
   end
+
+  # New method created in order to show the rates in a easier to understand way
+  def show_rates
+    rate = self.rate.to_s
+    return detail_currency.code if rate.to_f.zero?
+    per_time_map = {
+      :hourly  =>"Hour",
+      :daily   => "Day",
+      :weekly  => "Week",
+      :monthly => "Month",
+      :yearly  => "Year"
+    }
+    hr = per_time_map[per_time.to_sym]
+    raise "rate time unit of '#{per_time}' not supported" if hr.nil?
```

Comment:
where is per_time coming from?  How can something but hourly, daily, weekly, monthly, or yearly be chosen?  I think we should either validate in their selection at selection time or via model validation.    That would make this nil check unneeded.